dashboard/src/components/AppView/AppView.tsx

```diff
@@ -136,7 +136,7 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
         this.setState({ deployments: { ...this.state.deployments, [key]: resource } });
         break;
       case "Service":
-        this.setState({ services: { ...this.state.services, [key]: resource } });
+        this.setState({ services: this.state.services.set(key, resource) });
```

Comment:
So we were transforming the map into an object? Is the change to preserve the type? Why did you change it?